circom-types/src/traits.rs

```diff
@@ -512,10 +522,30 @@ where
     const GT_SERIALIZED_BYTE_SIZE_COMPRESSED: usize;
     /// Size of uncompressed element of Gt in bytes
     const GT_SERIALIZED_BYTE_SIZE_UNCOMPRESSED: usize;
+    /// Deserializes element of G1 from bytes where the element is already in montgomery form (no montgomery reduction performed)
+    fn g1_from_bytes(bytes: &[u8]) -> IoResult<Self::G1Affine>;
+    /// Deserializes element of G2 from bytes where the element is already in montgomery form (no montgomery reduction performed)
+    fn g2_from_bytes(bytes: &[u8]) -> IoResult<Self::G2Affine>;
     /// Deserializes element of G1 from reader where the element is already in montgomery form (no montgomery reduction performed)
     fn g1_from_reader(reader: impl Read) -> IoResult<Self::G1Affine>;
     /// Deserializes element of G2 from reader where the element is already in montgomery form (no montgomery reduction performed)
     fn g2_from_reader(reader: impl Read) -> IoResult<Self::G2Affine>;
+    /// Deserializes vec of G1 from reader where the elements are already in montgomery form (no montgomery reduction performed)
+    fn g1_vec_from_reader(mut reader: impl Read, num: usize) -> IoResult<Vec<Self::G1Affine>> {
```

Comment:
Two things

- write in docs that the implementation uses rayon and that it is multithreaded
- also I do not know if this function has a better place in zkey groth16 and not on the trait

Reply:
same for the g2 obv